Neos.Cache/Classes/Backend/MultiBackend.php

```diff
@@ -15,57 +15,48 @@
 
 use Neos\Cache\BackendInstantiationTrait;
 use Neos\Cache\EnvironmentConfiguration;
+use Neos\Flow\Core\Bootstrap;
+use Neos\Flow\Log\ThrowableStorageInterface;
+use Neos\Flow\Log\Utility\LogEnvironment;
+use Neos\Flow\ObjectManagement\ObjectManagerInterface;
+use Psr\Log\LoggerInterface;
+use Throwable;
 
 /**
  * A multi backend, falling back to multiple backends if errors occur.
  */
-class MultiBackend extends AbstractBackend implements BackendInterface
+class MultiBackend extends AbstractBackend
 {
     use BackendInstantiationTrait;
 
-    /**
-     * Configuration for all sub backends (each with the keys "backend" and "backendOptions")
-     *
-     * @var array
-     */
-    protected $backendConfigurations = [];
-
-    /**
-     * @var BackendInterface[]
-     */
-    protected $backends = [];
-
-    /**
-     * @var bool
-     */
-    protected $setInAllBackends = true;
-
-    /**
-     * @var bool
-     */
-    protected $debug = false;
+    protected array $backendConfigurations = [];
+    protected array $backends = [];
+    protected bool $setInAllBackends = true;
+    protected bool $debug = false;
+    protected bool $logErrors = true;
+    protected bool $removeUnhealthyBackends = true;
+    protected bool $initialized = false;
 
-    /**
-     * Are the backends initialized
-     *
-     * @var bool
-     */
-    protected $initialized = false;
+    protected ?LoggerInterface $logger = null;
+    protected ?ThrowableStorageInterface $throwableStorage = null;
 
-    /**
-     * Constructs this backend
-     *
-     * @param EnvironmentConfiguration $environmentConfiguration
-     * @param array $options Configuration options - depends on the actual backend
-     */
-    public function __construct(EnvironmentConfiguration $environmentConfiguration, array $options)
+    public function __construct(EnvironmentConfiguration $environmentConfiguration = null, array $options = [])
     {
         parent::__construct($environmentConfiguration, $options);
+
+        /** @noinspection ClassConstantCanBeUsedInspection */
+        if ($this->logErrors && class_exists('Neos\Flow\Core\Bootstrap') && Bootstrap::$staticObjectManager instanceof ObjectManagerInterface) {
```

Comment:
Why not use the `::class` constant here? Any special reason?
```suggestion
        if ($this->logErrors && class_exists(Bootstrap::class) && Bootstrap::$staticObjectManager instanceof ObjectManagerInterface) {
```

Reply:
I always forget that `::class` is even allowed for classes which don't exist (which is somewhat confusing). Fixed that.